src/components/common/BaseDatasourceSettings.tsx

```diff
@@ -0,0 +1,109 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+
+import Card from 'react-md/lib/Cards/Card';
+import TextField from 'react-md/lib/TextFields';
+import FontIcon from 'react-md/lib/FontIcons';
+import CardTitle from 'react-md/lib/Cards/CardTitle';
+import Switch from 'react-md/lib/SelectionControls/Switch';
+import Avatar from 'react-md/lib/Avatars';
+import SelectField from 'react-md/lib/SelectFields';
+
+import SettingsActions from '../../actions/SettingsActions';
+
+export interface IBaseSettingsProps {
+  settings: DataSource;
+}
+
+export interface IBaseSettingsState { 
+}
+
+export abstract class BaseDatasourceSettings<T extends IBaseSettingsState> 
+                      extends React.Component<IBaseSettingsProps, T> {
+
+  // require derived classes to implement
```

Comment:
This comment is obvious (since it's an abstract) should be something like 
`// an icon to be displayed when selecting the specific data source settings in the settings dialog`

Also add a comment on renderChildren